data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java

```diff
@@ -0,0 +1,128 @@
+    this.syncDataProvider = dataProvider.getSyncDataProvider();
+    this.jsonProvider = jsonProvider;
+  }
+
+  PostBlock(
+      final ValidatorDataProvider validatorDataProvider,
+      final SyncDataProvider syncDataProvider,
+      final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.validatorDataProvider = validatorDataProvider;
+    this.syncDataProvider = syncDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.POST,
+      summary = "Publish a signed block",
+      tags = {TAG_V1_BEACON, TAG_VALIDATOR_REQUIRED},
+      requestBody =
+          @OpenApiRequestBody(content = {@OpenApiContent(from = SignedBeaconBlock.class)}),
+      description =
+          "Submit a signed beacon block to the beacon node to be imported."
+              + " The beacon node performs the required validation.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            description = "Block has been successfully broadcast, validated and imported."),
+        @OpenApiResponse(
+            status = RES_ACCEPTED,
+            description =
+                "Block has been successfully broadcast, but failed validation and has not been imported."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Unable to parse request body."),
+        @OpenApiResponse(
+            status = RES_INTERNAL_ERROR,
+            description = "Beacon node experienced an internal error."),
+        @OpenApiResponse(
+            status = RES_SERVICE_UNAVAILABLE,
+            description = "Beacon node is currently syncing.")
+      })
+  @Override
+  public void handle(final Context ctx) throws Exception {
+    try {
+      if (syncDataProvider.getSyncStatus().is_syncing) {
+        ctx.status(SC_SERVICE_UNAVAILABLE);
+        return;
+      }
+
+      final SignedBeaconBlock signedBeaconBlock =
+          jsonProvider.jsonToObject(ctx.body(), SignedBeaconBlock.class);
+
+      ctx.result(
+          validatorDataProvider
+              .submitSignedBlock(signedBeaconBlock)
+              .thenApplyChecked(
+                  validatorBlockResult -> handleResponseContext(ctx, validatorBlockResult)));
+
+    } catch (final JsonMappingException | JsonParseException ex) {
+      ctx.status(SC_BAD_REQUEST);
+    } catch (final Exception ex) {
+      ctx.status(SC_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private String handleResponseContext(
+      final Context ctx, final ValidatorBlockResult validatorBlockResult)
+      throws JsonProcessingException {
+    ctx.status(validatorBlockResult.getResponseCode());
+    if (validatorBlockResult.getFailureReason().isPresent()) {
+      return jsonProvider.objectToJSON(validatorBlockResult.getFailureReason().get());
+    } else {
+      return jsonProvider.objectToJSON(
+          validatorBlockResult.getHash_tree_root().map(Bytes32::toHexString).orElse(""));
+    }
```

Comment:
I don't think this matches the expected output.  It looks like the new API has no content for either 200 or 202 and error responses should have a code and message.

Reply:
Can we also add some tests to cover what this should output please? I discovered on the weekend we had none and it was actually converting an `Optional` to JSON which wasn't pretty.

Reply:
Hrm I just lifted and shifted, i'll take another pass.